src/analyses/guard_bdd.cpp

```diff
@@ -0,0 +1,306 @@
+/*******************************************************************\
+
+Module: Guard Data Structure
+
+Author: Romain Brenguier, romain.brenguier@diffblue.com
+
+\*******************************************************************/
+
+/// \file
+/// Implementation of guards using BDDs
+
+#include "guard_bdd.h"
+
+#include <ostream>
+
+#include "expr_util.h"
+#include "invariant.h"
+#include "make_unique.h"
+#include "namespace.h"
+#include "simplify_utils.h"
+#include "std_expr.h"
+#include "symbol_table.h"
+#include <algorithm>
+#include <set>
+#include <solvers/bdd/bdd.h>
+
+#include "range.h"
+
+static std::size_t var_of_expr(guard_bdd_managert &manager, const exprt &expr)
+{
+  auto entry = manager.expr_to_var.insert(
+    std::make_pair(expr, manager.var_to_expr.size()));
+  if(entry.second)
+    manager.var_to_expr.push_back(expr);
+  return entry.first->second;
+}
+
+/// Convert an expression to a bddt
+static bddt bdd_from_expr(const exprt &expr, guard_bdd_managert &manager)
```

Comment:
How does this relate/compare to `solvers/prop/bdd_expr.{h,cpp}`?

Reply:
Indeed there are very similar. Originaly the bdd_expr implementation was for miniBDD and this one for Cudd but now that they are both based on the unified interface the `from_expr` and `as_expr` are basically duplicates. I could move the bdd_expr from solvers/prop to analyses and use that in guard_bdd, but I think the class needs to be adapted a bit, in particular the `bdd_exprt` shouldn't own the bdd_manager and the expr to bdd map.

Reply:
sorry for the confusing I, don't need to move bdd_expr since analyses can use solvers. I will just add to the interface of `bdd_expr` to expose `from_expr_rec` and `as_expr` as non member functions, and use these directly from `bdd_guard`

Reply:
Ok, whatever helps to avoid having two almost-the-same implementations seemingly solving the same problem. I don't mind you replacing my past implementation, I just want to make sure we don't have duplicate code.